packages/hydrogen/src/Numbers/UIAnimatedNumber.tsx

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import { View, Platform, StyleSheet } from 'react-native';
+import Animated, {
+    Easing,
+    useAnimatedProps,
+    useDerivedValue,
+    useSharedValue,
+    withTiming,
+} from 'react-native-reanimated';
+import { uiLocalized } from '@tonlabs/uikit.localization';
+
+import { Typography } from '../Typography';
+
+import { AnimatedTextInput } from './AnimatedTextInput';
+import { useNumberValue } from './useNumberValue';
+import {
+    runOnUILocalizedNumberFormat,
+    UINumberDecimalAspect,
+} from './runOnUILocalizedNumberFormat';
+import { useNumberStaticStyles } from './UIStaticNumber';
+import type { UINumberAppearance, UINumberGeneralProps } from './types';
+import { useTextLikeContainer } from './useTextLikeContainer';
+import { DebugGrid } from './DebugGrid';
+
+Animated.addWhitelistedNativeProps({ text: true });
+
+export function UIAnimatedNumber({
+    testID,
+    children,
+    decimalAspect = UINumberDecimalAspect.None,
+    // appearance
+    integerVariant,
+    integerColor,
+    decimalVariant,
+    decimalColor,
+    sign,
+    showDebugGrid,
+    showPositiveSign,
+}: UINumberGeneralProps & UINumberAppearance & { sign?: React.ReactNode }) {
+    const value: number = useNumberValue(children, decimalAspect);
```

Comment:
I'm scared with the fact that we are to convert BigNumbers to Numbers here, since we can easily loose the precision for nano tokens

I suggest an idea to parse the BigNumber and get its `integer` and `decimal` parts as Numbers (they must fit `MAX_SAFE_INTEGER`) and hence we'll be able to avoid any issues.

```suggestion
    const { integer, decimal } = useNumberValue(children, decimalAspect);
```